attribute.cpp

```diff
@@ -155,9 +164,69 @@ void preserve_attribute(attribute_op const &op, std::string const &key, serial_v
 					state->second.count += 1;
 					full_values[i].s = std::to_string(state->second.count);
 				}
-				break;
+				return;
 			}
 			}
 		}
 	}
+
+	// not found, so we are making a new value
+
+	serial_val sv;
+	switch (op) {
+	case op_sum:
+	case op_product:
+	case op_max:
+	case op_min:
+		sv.s = val.s;
+		sv.type = mvt_double;
+		break;
+
+	case op_mean: {
+		auto state = attribute_accum_state.find(key);
+		if (state == attribute_accum_state.end()) {
+			accum_state s;
+			s.sum = atof(val.s.c_str());
+			s.count = 1;
+			attribute_accum_state.insert(std::pair<std::string, accum_state>(key, s));
+
+			sv.s = milo::dtoa_milo(s.sum / s.count);
+		} else {
+			state->second.sum += atof(val.s.c_str());
+			state->second.count += 1;
+
+			sv.s = milo::dtoa_milo(state->second.sum / state->second.count);
+		}
+		break;
+	}
+
+	case op_concat:
+		sv.s = val.s;
```

Comment:
Let's make these cases that can't happen raise an error instead of having semi-speculative (probably correct!) implementations